doc/arch/adr-003.md

```diff
@@ -0,0 +1,51 @@
+# ADR 003: New splits will be prefixed with `[app_name].`
+
+## Status
+
+Accepted
+
+## Context
+
+In order to allow apps to ship independently, we need to ensure that
+split names defined in different apps don't collide. The strategy until
+now has been for developers to run all the apps that might have
+conflicting splits in local dev, but this doesn't scale well as they may
+not have the latest version of each app at all times and a
+fully-upgraded local testtrack server to detect conflicts that might
+arise in production.
+
+We need to make sure that migrations don't get jammed up in production
+even if a developer only has their own app downloaded locally.
+
+## Decision
+
+New migration runners will be expected to prefix their split names with
```

Comment:
Maybe we're too far down one path for this to make sense, but I'd be interested in your thoughts on allowing apps to have a local scope on split names, where they don't need to set their own prefix (or even see it on lookups). They'd only see prefixes for splits owned by _other_ apps.

Reply:
> Yup! That is exactly where we’re headed! The wire protocol will be fully qualified, but the migrations and client variant/AB calls will default to local scope.

Oh, nice. Then carry on 😉